### DISCUSSION TITLE
Can we make switch-preset-column-width an action?

### DISCUSSION THREAD

**Discussion by kls1991** [YaLTeR/niri#470] [Ideas]
I found my self only use a few window widths: 1/3, 1/2, 2/3, 5/6, and fullscreen.

So in theory I can only use two shortcuts to ajust width efficiently(three if Mod+F counted)

I consider something like this

```kdl
binds {
    // switch between [S]mall widths
    Mod+S { switch-column-width "33.3333%" "50%"; }

    // switch between [G]iant widths
    Mod+G { switch-column-width "66.6667%" "83.3333%"; }
}
```

I'm so proud of this idea: Say I want a window to be big or small, but not sure how big/small it should be exactly. Just hit the shortcuts one or two times, it will be ok.

------------------------------
### Added
Ignore me please. I did some testing it doesn't feel natrue as I thought this way.
And I realized a window's width will not change from very small to very big often.
I'm going to try make Mod+Minus/Equal step by 1/6 for some days.

**Comment by YaLTeR** [comment]
While possible, this would be pretty annoying to add.

**Reply by kls1991** [reply]
> While possible, this would be pretty annoying to add.

Forget about my stupid idea yesterday. I just change my keybinds and feel great

```kdl
binds {
    // N for narrow
    Mod+N     { set-column-width "-16.6667%"; }
    // W for wide
    Mod+W    { set-column-width "+16.6667%"; }
    // M for medium
    Mod+M    { set-column-width "66.6667%"; }
}
```
